### DISCUSSION TITLE
Whether is it support nested visualization?

### DISCUSSION THREAD

**Discussion by ByteFlyCoding** [lima-vm/lima#1777] [Q&A] [answered]
I want create a virtual machine in a existing virtual machine that run on macbook pro of apple sillicon m1 pro. This may need support nested visualization. But I don't know lima-vm whether it is support nested visualization.

**Comment by AkihiroSuda** [comment]
Nested virtualization seems missing on the M1 series.

[Apparently M2 supports nested virt](https://social.treehouse.systems/@marcan/109838053800961073), but Lima can't support it until Virtualization.framework or QEMU supports it.

**Comment by msgilligan** [comment]
Virtualization Framework beta documentation says it will work on macOS 15+ on M3+:
https://developer.apple.com/documentation/virtualization/vzgenericplatformconfiguration/4360553-isnestedvirtualizationsupported?changes=latest_minor

**Comment by abiosoft** [comment]
I can confirm that I successfully tested nested virtualization support on m3 Macbook running macOS 15 beta and `kvm-ok` does indeed report success :tada:.

**Comment by asauber** [comment]
@abiosoft could you share which distro image and lima configuration that you used for testing? With some brief testing, I haven't yet been able to see `/dev/kvm` on a guest.

**Reply by abiosoft** [reply]
I used Ubuntu 24.04

**Reply by abiosoft** [reply]
It's not yet merged in. 
You can check https://github.com/lima-vm/lima/pull/2530#issuecomment-2335110455 for context.

**Reply by asauber** [reply]
I was able to create a Lima VM with KVM support with your testing instructions.

```
$ limactl shell ubuntunested
WARN[0000] treating lima version "bc39aed.m" from "/Users/asauber/.lima/ubuntunested/lima-version" as very latest release
asauber@lima-ubuntunested:/Users/asauber$ kvm-ok
INFO: /dev/kvm exists
KVM acceleration can be used
```

Very cool. Thank you for working on this!

**Reply by AnthoBalitrand** [reply]
Hello guys, 
Can you share the details of the config you used to make nested virtualization work ? 
(Mainly vmType value and exact image name, along with other non-default parameters you set). 

With nestedVirtualization: true, vmType: vz and image 24.10/release-20241212 I do not manage to make it work on MacBook Pro M3 running MacOS 15.2.

**Reply by AkihiroSuda** [reply, marked as answer]
```
limactl start --set '.nestedVirtualization=true' template://default
```

See also:
- https://github.com/lima-vm/lima/pull/3061/files

**Reply by AnthoBalitrand** [reply]
Still not working : 

```
INFO[0001] Starting the instance "default" with VM driver "vz"
INFO[0001] Attempting to download the image              arch=aarch64 digest="sha256:fb39312ffd2b47b97eaef6ff197912eaa3e0a215eb3eecfbf2a24acd96ee1125" location="https://cloud-images.ubuntu.com/releases/24.10/release-20241212/ubuntu-24.10-server-cloudimg-arm64.img"
INFO[0001] Using cache "/Users/xxx/Library/Caches/lima/download/by-url-sha256/e540d15172f1f5be55c9c20dc5bc7ed4338279091f52a8f468a68363673c7547/data"
INFO[0001] Converting "/Users/xxx/.lima/default/basedisk" (qcow2) to a raw disk "/Users/xxx/.lima/default/diffdisk"
3.50 GiB / 3.50 GiB [---------------------------------------] 100.00% 1.55 GiB/s
INFO[0003] Expanding to 100GiB
INFO[0003] Attempting to download the nerdctl archive    arch=aarch64 digest="sha256:af60f6b83074c319cf1d0264b94a2a2f78cb757fe296f8193867d1cf79956e71" location="https://github.com/containerd/nerdctl/releases/download/v2.0.1/nerdctl-full-2.0.1-linux-arm64.tar.gz"
INFO[0003] Using cache "/Users/xxx/Library/Caches/lima/download/by-url-sha256/35c5b08543fe221085711b99a84cac8c496f73d049c8c8795ba0b6b2faac7b5f/data"
INFO[0003] [hostagent] hostagent socket created at /Users/xxx/.lima/default/ha.sock
INFO[0003] [hostagent] Starting VZ (hint: to watch the boot progress, see "/Users/xxx/.lima/default/serial*.log")
FATA[0004] exiting, status={Running:false Degraded:false Exiting:true Errors:[] SSHLocalPort:0} (hint: see "/Users/xxx/.lima/default/ha.stderr.log")
```

ha.stderr.log contains the following : 

`{"level":"fatal","msg":"nested virtualization is not supported on this device","time":"2025-01-27T14:09:28+01:00"}`

Is there any MacOS command (sysctl ?) to check if nested virtualization has been disabled by any organization profile ?
